### ISSUE TITLE
Please document how to retain message order when producing messages via a Broker

### ISSUE THREAD

**Issue by FWiesner** [knative/docs#2198]
**Describe the change you'd like to see**
Kafka at least offers the capability to set the partition key to ensure that messages with the same key make it to the same topic partition and thereafter get delivered in sequence of their insertion.
I do not know whether it is a documentation or feature request (or both). But, I doubt that there is no implementation yet and would like to understand from the docs, how to set the partition key when calling the Broker for message creation.

**Additional context**

**Comment by samodell** [comment]
Adding @vaikas  to take a look from the eventing perspective -- feel free to reassign to someone else.

**Comment by FWiesner** [comment]
This directly relates to https://github.com/knative/eventing-contrib/issues/944

**Comment by vaikas** [comment]
TL;DR, today this is not possible. There's some ongoing work to figure out how we might be able to change things.
